### PR TITLE
Feature: Authentication

### PR DESCRIPTION
**Pull request by anandsr21harshit**
- Added mockbee authentication
- Created login page

### CODE REVIEW

File: keep-notes/src/context/auth-context.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { createContext, useContext, useState } from "react";
+import axios from "axios"
+
+const AuthContext = createContext();
+
+const AuthProvider = ({children})=>{
+    const localStorageToken = JSON.parse(localStorage.getItem("loginCred"))
+    console.log(localStorageToken);
```
Comment (manish591) [inline keep-notes/src/context/auth-context.jsx]: Remove console

File: keep-notes/src/context/auth-context.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            console.log(response);
+            if(response.status === 200 || response.status === 201){
+                localStorage.setItem("loginCred",JSON.stringify({token: response.data.encodedToken,user:response.data.foundUser}));
+                setCurrentUser(response.data.foundUser);
+                setToken(response.data.encodedToken);
+            }
+            console.log("logged in");
+        }
+        catch(err){
+            console.log(err);
```
Comment (manish591) [inline keep-notes/src/context/auth-context.jsx]: Use console.error for error message

**Review by manish591 — CHANGES_REQUESTED** [review]
Code is looking good, here are few suggestions
* remove unnecessary console message,
* use prettier for consistent formatting
* use barrel imports for better readability of import statements

File: keep-notes/src/context/auth-context.jsx (hunk 1 above)
Comment (anandsr21harshit) [reply keep-notes/src/context/auth-context.jsx]: Hi Manish,
I have removed irrelevant console log, added barrel import and used console.error() whereever required.
Please review it.

**Review by anuj-kumary — APPROVED** [review]
Clean code
Good naming conventions
Nice logic applied

File: keep-notes/src/App.js [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
-import {Routes, Route} from "react-router-dom"
-
+import {
+  LandingPage,
+  Home,
+  Labels,
+  Archive,
+  Trash,
+  Login,
+} from "./pages/index";
```
Comment (ramalingaa) [inline keep-notes/src/App.js:8]: Good usage of index imports

File: keep-notes/src/components/NotesInput.jsx [hunk 4]
```diff
@@ -1,33 +1,102 @@
...
+  };
+
+  const [note, setNote] = useState(initialNote);
+
+  const notesHandler = async () => {
+    try{
+      const response = await axios.post("/api/notes",note,{headers:{authorization: token}})
+
+      if(response.status === 200 || response.status === 201){
+        console.log("check response", response.data.notes)
```
Comment (ramalingaa) [inline keep-notes/src/components/NotesInput.jsx:33]: Please remove console

File: keep-notes/src/components/NotesInput.jsx [hunk 5]
```diff
@@ -1,33 +1,102 @@
...
-      </section>
-      <footer className='notes-footer'>
+      <div className="notes-input-container">
+        <header className="notes-header">
+          <input
+            className="notes-title"
+            type="text"
+            placeholder="Title"
+            value={note.title}
+            onChange={(e) => setNote({ ...note, title: e.target.value })}
```
Comment (ramalingaa) [inline keep-notes/src/components/NotesInput.jsx:51]: Try to use updater function for state updates

**Review by ramalingaa — APPROVED** [review]
Code looks clean, please make changes before merging.

File: keep-notes/src/components/NotesInput.jsx (hunk 5 above)
Comment (anandsr21harshit) [reply keep-notes/src/components/NotesInput.jsx:51]: creating a new branch for adding notes will surely make changes there.

**Pull request merged**